### ISSUE TITLE
Distortion appearing using affine for build_template.

### ISSUE THREAD

**Issue by Nicksolas** [rohitrango/FireANTs#64]
Hey there! Just found this awesome Project and am trying to use the build template function. 
The function works well as long as i don't use the affine transform.  When I use it the output becomes very distorted and clusters in the middle of the volume.
Am i doing something wrong or might be there an issue?

Here is the options i am using:
debug: True
verbose: True

# laplacian params
num_laplacian: 2
laplace_params:
  learning_rate: 0.5
  itk_scale: true

# shape averaging
shape_avg: false

# registration params
batch_size: 1

do_moments: False
moments:
  scale: 4
  moments: 1

# rigid transform
do_rigid: True
rigid:
  iterations: [100, 100, 60, 20]
  scales: [8, 4, 2, 1]
  progress_bar: ${verbose}
  loss_type: cc

# affine transform
do_affine: True
affine:
  iterations: [100, 100, 60, 20]
  scales: [8, 4, 2, 1]
  progress_bar: ${verbose}
  loss_type: cc

# deformation transform
do_deform: True
deform_algo: greedy
deform:
  iterations: [100, 100, 60, 40]
  scales: [8, 4, 2, 1]
  optimizer_lr: 0.05        # for template creation, use lower learning rate than pairwise
  smooth_warp_sigma: 0.5
  smooth_grad_sigma: 1
  progress_bar: ${verbose}
  cc_kernel_size: 5
  loss_type: fusedcc

# additional params
tmpdir: null
last_reg: null

**Comment by JillSunday** [comment]
Hi @rohitrango and @Nicksolas ,
I am trying to use FireANTs on a template building project. I seem to be running into a similar problem. My templates appear to be more distorted over iterations. template_0 comes out perfect but the others appear deformed, so to speak. The same goes for the moved images, per iteration. The moved images from the first iteration are okay, template okay, but not for the other iterations.

I changed the scales and iterations after looking at this issue. I thought that might be the problem but nothing has changed. @Nicksolas , did you manage to make it work on your dataset? And @rohitrango , could you please advise on where I might be going wrong?  Here is a copy of my config: 

debug: True
verbose: True

image_list_file: all_images.txt 
image_prefix: null
image_suffix: null
num_subjects: null

#Template details
init_template_path: MNI_brain.nii.gz
template_iterations: 6
normalize_images: False
save_as_uint8: False

# Laplacian regularization
num_laplacian: 2
laplace_params:
  learning_rate: 0.5
  itk_scale: true

# Shape averaging
shape_avg: false

#output
save_dir: ./template_iterations 
save_every: 1
save_init_template: true
save_moved_images: true
save_additional:
  image_file: []
  image_prefix: []
  image_suffix: []
  is_segmentation: []

#registration params
batch_size: 1

do_moments: false
moments:
  scale: 4
  moments: 2
  orientation: rot
  loss_type: fusedcc
  perform_scaling: true

#rigid transform
do_rigid: true
rigid:
  #iterations: [200,100,25]
  iterations: [100,100,60,20]
  #scales: [4,2,1]
  scales: [8,4,2,1]
  progress_bar: ${verbose}
  loss_type: fusedcc

#affine transform
do_affine: true
affine:
  #iterations: [200,100,50]
  iterations: [100,100,60,20]
  #scales: [4,2,1]
  scales: [8,4,2,1]
  progress_bar: ${verbose}
  #optimizer: Adam
  #optimizer_lr: 0.03
  #cc_kernel_size: 5
  loss_type: fusedcc

#deformation transform
do_deform: true
deform_algo: greedy
deform:
  #iterations: [200,100,50]
  iterations: [100,100,60,40]
  #scales: [4,2,1]
  scales: [8,4,2,1]
  #optimizer_lr: 0.25
  optimizer_lr: 0.5
  smooth_warp_sigma: 0.5
  smooth_grad_sigma: 1.0
  progress_bar: ${verbose}
  cc_kernel_size: 5
  loss_type: fusedcc

#additional params
tmpdir: null
last_reg: null
